pkg/crypto/tinkcrypto/key_wrapper.go

```diff
@@ -516,12 +518,12 @@ func (t *Crypto) deriveESWithOKPKeyForUnwrap(alg string, apu, apv []byte, epk *c
 	epkChacha := new([chacha20poly1305.KeySize]byte)
 	copy(epkChacha[:], epk.X)
 
-	kek, err := cryptoutil.Derive25519KEK([]byte(alg), apu, apv, recPrivKeyChacha, epkChacha)
+	z, err := cryptoutil.DeriveECDHX25519(recPrivKeyChacha, epkChacha)
 	if err != nil {
-		return nil, fmt.Errorf("deriveESWithOKPKeyForUnwrap: failed to derive kek: %w", err)
+		return nil, fmt.Errorf("deriveESWithOKPKeyForUnwrap: %w", err)
 	}
 
-	return kek, nil
+	return kdf(alg, z, apu, apv, chacha20poly1305.KeySize), nil
```

Comment:
What about the AES version?

Reply:
AES uses the new [DeriveECDH](https://github.com/hyperledger/aries-framework-go/pull/2767/files/45126548a79b1df9a6f3e1826a76229844cf372c#diff-afa818d584f27877524fa05cc089be42b2857a371576390df417bb9680bbdb8aR130) call below